src/app/login/login-form.directive.ts

```diff
@@ -0,0 +1,12 @@
+import { Directive, ElementRef } from '@angular/core';
+
+@Directive({
+    selector: '[lowercase]'
+})
+export class LowerCase {
+  constructor(public el: ElementRef) {
+    this.el.nativeElement.onkeypress = (evt) => {
+      (evt.which < 65 || evt.which > 90) ? "" : evt.preventDefault();
```

Comment:
it only prevents key code. Actually we need to change it to lowecase, needn't we ?